pkg/network/infraconfigurators/bridge_test.go

```diff
@@ -158,4 +158,484 @@ var _ = Describe("Bridge infrastructure configurator", func() {
 			})
 		})
 	})
+
+	Context("prepre the pod networking infrastructure", func() {
+		const (
+			ifaceName   = "eth0"
+			launcherPID = 1000
+		)
+
+		var (
+			bridgeConfigurator *BridgePodNetworkConfigurator
+			bridgeIPAddr       *netlink.Addr
+			bridgeIPStr        string
+			iface              *v1.Interface
+			inPodBridge        *netlink.Bridge
+			mac                net.HardwareAddr
+			mtu                int
+			podLink            *netlink.GenericLink
+			podIP              netlink.Addr
+			queueCount         uint32
+			tapDeviceName      string
+			vmi                *v1.VirtualMachineInstance
+		)
+
+		BeforeEach(func() {
+			iface = v1.DefaultBridgeNetworkInterface()
+			vmi = newVMIBridgeInterface("default", "vm1")
+
+			bridgeConfigurator = NewBridgePodNetworkConfigurator(vmi, iface, bridgeIfaceName, launcherPID, handler)
+		})
+
+		BeforeEach(func() {
+			macStr := "AF:B3:1F:78:2A:CA"
+			mac, _ = net.ParseMAC(macStr)
+			mtu = 1000
+			podLink = &netlink.GenericLink{LinkAttrs: netlink.LinkAttrs{Name: ifaceName, MTU: mtu}}
+			queueCount = uint32(0)
+			tapDeviceName = "tap0"
+		})
+
+		When("the pod features an L3 configured network (IPAM)", func() {
```

Comment:
What do you think about transforming it into a DescribeTable with a a struct with one of the arguments a function to setup handler expeactations ? 

```golang
type prepareNetworkingIPAMCase struct {
    handlerExpectations func(h *MockNetworkHandler)
    should GinkgoMatcher
}

Entry("when all network drivers calls succeed", prepareNetworkingIPAMCase{
   handlerExpectations: func(h *MockNetworkhandler) {
      h.EXPECT()...
      ...
   }
   should: Succeed()
}
```

Reply:
I think that's an amazing opportunity for a follow-up.